aws/cloudformation/components/ami.yml.erb

```diff
@@ -1,4 +1,40 @@
-<% ami = commit[0..4] -%>
+<%
+# Template component to deploy AMI-backed frontends to an autoscaling group with rolling updates.
+
+# String to append to AMI logical resources that get replaced on every build.
+# Typically a portion of the commit hash.
+ami ||= raise('ami string required')
+
+# Shell script to run when building/updating the AMI-creating instance.
+build_ami ||= <<SH
+INSTANCE_ID=$(curl -s http://169.254.169.254/latest/meta-data/instance-id)
+STACK=${AWS::StackName}
+RESOURCE_ID=AMICreate#{ami}
+STATUS=SUCCESS
+REGION=${AWS::Region}
+SH
+
+# Shell script to run when building/updating each frontend instance.
+build_frontend ||= <<SH
+sleep 10
```

Comment:
When testing on the minimal test stack, I found that sometimes a frontend would boot up so quickly that the lifecycle-hook endpoint hadn't been fully provisioned yet, so the call to `complete-lifecycle-action` would return an error along the lines of 'invalid instance ID for lifecycle action'. Didn't seem to affect the real stack since it was doing enough in its build script. Adding a little sleep fixed it for now.